task7/task7.js

```diff
@@ -0,0 +1,348 @@
+    
+    _posts = [];
+
+    constructor(posts) {
+        this._posts = posts;
+    }
+
+    static validate(post) {
+        if (!post) {
+            return false;
+        }
+
+        const { description, id, createdAt, author, photoLink='' } = post;
+        const requiredFieldsExist = id
+            && description
+            && createdAt
+            && author
+
+        const idIsValid = typeof id === 'string' && !getPost(id);
+        const descriptionIsValid = typeof description === 'string' && MAX_DESCRIPTION_LENGTH < 200;
+        const createdAtIsValid = createdAt instanceof Date && !isNaN(createdAt.getTime());
+        const authorIsValid = typeof author === 'string' && author;
+        const photoLinkIsValid = typeof photoLink === 'string';
+
+        return requiredFieldsExist
+            && idIsValid
+            && descriptionIsValid
+            && createdAtIsValid
+            && authorIsValid
+            && photoLinkIsValid;
+    }
+
+    getPage(skip = DEFAULT_SKIP, top = DEFAULT_TOP, filterConfig = {}) {
+        if (skip < 0 || skip > this._posts.length) {
+            return [];
+        }
+
+        if (top < 0) {
+            return [];
+        }
+
+        const slicedPosts = this._posts.sort((firstPost, secondPost) => secondPost.createdAt - firstPost.createdAt).slice(skip, top + skip);
+
+        if (typeof filterConfig !== 'object') {
+            return slicedPosts;
+        }
+
+        return slicedPosts.filter((post) => {
+            return (!filterConfig.id || filterConfig.id === post.id)
+                && (!filterConfig.description || filterConfig.description === post.description)
+                && (!filterConfig.createdAt || filterConfig.createdAt.getTime() === post.createdAt.getTime())
+                && (!filterConfig.author || filterConfig.author === post.author);
+        })
+    }
+
+    add(post) {
+        if (!this.constructor.validate(post)) {
+            return false;
+        }
+
+        this._posts.push(post);
+
+        return true;
+    }
+
+    edit(id, post) {
+        if (!post || !id) {
+            return false;
+        }
+
+        const currentPost = this.getPost(id);
+
+        if (!currentPost) {
+            return false;
+        }
+
+        const newPost = { ...currentPost, post };
+
+        if (!this.constructor.validate(currentPost)) {
+            return false;
+        }
+
+        this._posts[this._posts.indexOf(currentPost)] = newPost;
+        
+        return true;
+    }
+
+    remove(id) {
+        const post = this.getPost(id);
+
+        if (!post) {
+            return false;
+        }
+
+        return !!this._posts.slice(this._posts.indexOf(post), 1);
+    }
+
+    addAll(newPosts) {
+        return newPosts.reduce((invalidedPosts, newPost) => {
+            if (!this.add(newPost)) {
+                invalidedPosts = [...invalidedPosts, newPost];
+
+                return invalidedPosts;
+            }
+        }, [])
+    }
+
+
+    getPage() {

```

Comment:
Are you sure this is works :) 
I think no, keys (methods) are duplicated))

Reply:
yea
i duplicate only logs (they will be almost the same for everyone 👍 )